classes/order/Order.php

```diff
@@ -1097,6 +1098,36 @@ public static function getOrderByCartId($id_cart)
     }
 
     /**
+     * Get an order object by its cart id
+     *
+     * @param int $id_cart Cart id
+     * @return OrderCore
+     */
+    public static function getByCartId($id_cart)
+    {
+        $id_order = self::getIdByCartId($id_cart);
+
+        return new self($id_order);
```

Comment:
If there is no order with this cart id, I'm not sure you should return an empty order.